api/novelist.py

```diff
@@ -392,6 +391,23 @@ def process_item(self, identifier):
             self.output_source, metadata.primary_identifier,
             strength=1
         )
+        # Create an edition with the NoveList metadata and identifier
         edition, ignore = metadata.edition(self._db)
         metadata.apply(edition)
+        if edition.series or edition.series_position:
+            # Place any missing series data that NoveList gathered into this
+            # identifier's edition.
+            self._apply_series_data(
+                identifier, edition.series, edition.series_position
+            )
+
         return identifier
+
+    def _apply_series_data(self, identifier, series, series_position):
+        """Provides series data on the given identifier"""
+        licensed_edition = self.edition(identifier)
```

Comment:
I see what you're doing here but CoverageProvider.edition() is designed for CoverageProviders that are talking directly to the LicensePool source (e.g. the OverdriveBibliographicCoverageProvider). As such, it finds the Edition where the DataSource is the DataSource associated with the LicensePool. I think you want to create a new Edition where the DataSource is Novelist. 

You can do this by creating a Metadata object that includes primary_identifier, data_source, series_title, and series_position. Then call `edition()` on the Metadata object, and then `Metadata.apply(edition)`.

I'm not sure what to do about CoverageProvider.edition(), but it's going to need some work once it's possible to have multiple LicensePools for the same Identifier from different data sources.